common/lib/xmodule/xmodule/modulestore/__init__.py

```diff
@@ -169,6 +180,23 @@ class BulkOperationsMixin(object):
     def __init__(self, *args, **kwargs):
         super(BulkOperationsMixin, self).__init__(*args, **kwargs)
         self._active_bulk_ops = ActiveBulkThread(self._bulk_ops_record_type)
+        self._signal_handler = None
+
+    @property
+    def signal_handler(self):
+        """
+        Return a signal handler, defaults to a null handler that does nothing.
+        """
+        if not self._signal_handler:
+            self._signal_handler = NullSignalHandler()
+        return self._signal_handler
+
+    @signal_handler.setter
+    def signal_handler(self, value):
+        """
+        Set the signal handler
+        """
+        self._signal_handler = value
```

Comment:
Again, this `signal_handler` definition doesn't feel well-located when the rest of the signal-handling code is in `django.py`.